llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -718,6 +674,9 @@ void HWAddressSanitizer::initializeModule() {
   InstrumentGlobals =
       !CompileKernel && !UsePageAliases && optOr(ClGlobals, NewRuntime);
 
+  for (Function &F : M.functions())
+    removeASanIncompatibleFnAttributes(F, /*ReadsArgMem=*/UseShortGranules);
+
```

Comment:
Why move it here? 

IICU this now changes the behavior for HWSan, because we only drop write only now if `UseShortGranules`? If so, it needs a test (and might be better done separately)